### PR TITLE
HPCC-17293 Add an environment configuration (environment.conf) option to enable/disable DropZone restrictions.

### PR DESCRIPTION
**Pull request by AttilaVamos**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

**Comment by AttilaVamos** [comment]
The despray error was expected, because this PR enables the DropZone restriction, but the despray.ecl key file (intentionally) didn't changed.  (The persist_refresh in a known issue)

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -2705,7 +2705,9 @@ void FileSprayer::checkFilePath(RemoteFilename & filename)
             if (!dropZone)
             {
                 LOG(MCdebugInfo, unknownJob, "No matching drop zone path to file path: '%s'", pfilePath);
-                // throwError1(DFTERR_NoMatchingDropzonePath, pfilePath);
+                if (env->isDropZoneRestrictionEnabled())
+                    throwError1(DFTERR_NoMatchingDropzonePath, pfilePath);
```
Comment (jakesmith) [inline dali/ft/filecopy.cpp]: If not enabled, should still log.

File: initfiles/etc/DIR_NAME/environment.conf.in [hunk 2]
```diff
@@ -44,4 +44,8 @@ jvmoptions=-XX:-UsePerfData
 #
 # Multiple paths can be specified (separate with :, or ; on Windows).
 # Relative paths are assumed to be relative to ${INSTALL_DIR}/versioned
-additionalPlugins=python2
\ No newline at end of file
+additionalPlugins=python2
+
+# To en-/disable Drop Zone restriction.
+# Default is enable (true).
```
Comment (jakesmith) [inline initfiles/etc/DIR_NAME/environment.conf.in]: "enabled"

**Review by jakesmith — CHANGES_REQUESTED** [review]
@AttilaVamos - a couple of minor comments.

>The despray error was expected, because this PR enables the DropZone restriction, but the despray.ecl key file (intentionally) didn't changed.

Why (didn't you change the despray key) ?

File: dali/ft/filecopy.cpp (hunk 1 above)
Comment (AttilaVamos) [inline dali/ft/filecopy.cpp]: Changed

**Comment by AttilaVamos** [comment]
@jakesmith updated.

**Comment by jakesmith** [comment]
@AttilaVamos - what was the answer to https://github.com/hpcc-systems/HPCC-Platform/pull/9786#pullrequestreview-30233575 ?

**Comment by AttilaVamos** [comment]
I didn't change the key because I would like to test the feature.

**Comment by AttilaVamos** [comment]
The stepping7h (multiPart=true) error is parallel query related issue.

**Comment by jakesmith** [comment]
@richardkchapman - looks good to merge.

**Comment by richardkchapman** [comment]
I merged this then noticed that it needed squashing (2 of the commits are not properly formatted) so rolled back

**Comment by richardkchapman** [comment]
manually squashed and merged

**Comment by AttilaVamos** [comment]
@richardkchapman thank you.

**Pull request merged**